ipaserver/plugins/location.py

```diff
@@ -225,10 +225,17 @@ def execute(self, *keys, **options):
                     dns_servers.append(s_name)
 
             for server in servers_additional_info.values():
-                server['service_relative_weight'] = [
-                    u'{:.1f}%'.format(
-                        int(server['ipaserviceweight'][0])*100.0/weight_sum)
-                ]
+                if weight_sum != 0:
+                    server['service_relative_weight'] = [
+                        u'{:.1f}%'.format(
+                            int(server['ipaserviceweight'][0]) * 100.0 /
+                            weight_sum
+                        )
+                    ]
+                else:
+                    server['service_relative_weight'] = [
+                        u'{:.1f}%'.format(int(100.0 / len(servers)))
```

Comment:
It doesn't make sense to combine ``int()`` with float formatting. ``.1f`` formats the variable as float with one item after the dot, e.g. ``33.3%`` for 3 servers.